pandas-stubs/core/interchange/dataframe_protocol.pyi

```diff
@@ -0,0 +1,117 @@
+import abc
+from abc import (
+    ABC,
+    abstractmethod,
+)
+import enum
+from typing import (
+    Any,
+    Iterable,
+    Sequence,
+    TypedDict,
+)
+
+class DlpackDeviceType(enum.IntEnum):
+    CPU: int
+    CUDA: int
+    CPU_PINNED: int
+    OPENCL: int
+    VULKAN: int
+    METAL: int
+    VPI: int
+    ROCM: int
+
+class DtypeKind(enum.IntEnum):
+    INT: int
+    UINT: int
+    FLOAT: int
+    BOOL: int
+    STRING: int
+    DATETIME: int
+    CATEGORICAL: int
+
+class ColumnNullType(enum.IntEnum):
+    NON_NULLABLE: int
+    USE_NAN: int
+    USE_SENTINEL: int
+    USE_BITMASK: int
+    USE_BYTEMASK: int
+
+class ColumnBuffers(TypedDict):
+    data: tuple[Buffer, Any]
+    validity: tuple[Buffer, Any] | None
+    offsets: tuple[Buffer, Any] | None
+
+class CategoricalDescription(TypedDict):
+    is_ordered: bool
+    is_dictionary: bool
+    categories: Column | None
```

Comment:
Looking at https://data-apis.org/dataframe-protocol/latest/API.html, `categories` should be `dict | None`

Reply:
All of the types here are straight from pandas main.

Reply:
They may be wrong based on the docs, and are probably not tested within pandas itself, so can we fix here and do a PR there?

Reply:
Here https://github.com/pandas-dev/pandas/blob/396ad412cf4af59af199101d9b5e1e74f45c698a/pandas/core/interchange/dataframe_protocol.py#L273 it indicates that it should be a `Column`.

Reply:
I used this to verify the API:  https://data-apis.org/dataframe-protocol/latest/API.html

I think we may have errors in the pandas implementation _if_ that is the "official" standard.

Reply:
I suspect the code in pandas is the de facto even if it differs from the docs.

Reply:
I would also add that I think pandas-stubs should match pandas, and if there is an issue, pandas should change first, then pandas-stubs keep up.

Reply:
Did a little research, and it appears the docs I looked at are not up to date.  The interface changed in https://github.com/data-apis/dataframe-api/pull/69